it/src/test/java/com/linecorp/centraldogma/it/WatchTest.java

```diff
@@ -455,6 +498,222 @@ void transformingThread_withCustom(ClientType clientType) {
         await().untilAtomic(threadName, Matchers.startsWith(threadNamePrefix));
     }
 
+    @ParameterizedTest
+    @EnumSource(ClientType.class)
+    void fileWatcher_errorOnEntryNotFound(ClientType clientType)
+            throws Exception {
+        // Legacy client doesn't support this feature.
+        if (clientType == ClientType.LEGACY) {
+            return;
+        }
+
+        revertTestFiles(clientType);
+
+        // prepare test
+        final CentralDogma client = clientType.client(dogma);
+        final String filePath = "/test/not_exist.json";
+
+        // create watcher
+        final Watcher<JsonNode> watcher = client.fileWatcher(dogma.project(), dogma.repo1(),
+                                                             Query.ofJson(filePath), Function.identity(),
+                                                             blockingTaskExecutor, 100, true);
+
+        // check entry does not exist when to get initial value
+        assertThatThrownBy(watcher::awaitInitialValue)
+                .getRootCause().isInstanceOf(EntryNotFoundException.class);
+        assertThatThrownBy(() -> watcher.awaitInitialValue(100, TimeUnit.MILLISECONDS))
+                .getRootCause().isInstanceOf(EntryNotFoundException.class);
+        assertThatThrownBy(() -> watcher.awaitInitialValue(100, TimeUnit.MILLISECONDS, new TextNode("test")))
+                .getRootCause().isInstanceOf(EntryNotFoundException.class);
+
+        // when initialValueFuture throw 'EntryNotFoundException', you can't use 'watch' method.
+        assertThatThrownBy(() -> watcher.watch((rev, node) -> {
+        })).isInstanceOf(IllegalStateException.class);
+    }
+
+    @ParameterizedTest
+    @EnumSource(ClientType.class)
+    void fileWatcher_errorOnEntryNotFound_watch_method_is_not_working(
```

Comment:
Camel case is preferred. One underscore is acceptable to express different contexts.